### ISSUE TITLE
[jRuby 9.1.0.0] Strange behavior with Enumerable::max method

### ISSUE THREAD

**Issue by zocoi** [jruby/jruby#3856]
### Environment

jRuby 9.1.0.0

### Expected Behavior
in 9.0.1.0
```
jruby-9.0.1.0 :002 > [1,2,3,4,5].max {|el| puts el.inspect; el}
2
3
4
5
 => 5 
```

### Actual Behavior
```
jruby-9.1.0.0 :002 > [1,2,3,4,5].max {|el| puts el.inspect; el }
[2, 1]
NoMethodError: undefined method `>' for [2, 1]:Array
	from org/jruby/RubyArray.java:4211:in `max'
	from (irb):2:in `<eval>'
	from org/jruby/RubyKernel.java:983:in `eval'
	from org/jruby/RubyKernel.java:1290:in `loop'
	from org/jruby/RubyKernel.java:1103:in `catch'
	from org/jruby/RubyKernel.java:1103:in `catch'
```

### In MRI 2.3.1
```
2.3.1 :001 > [1,2,3,4,5].max {|el| puts el.inspect; el }
2
3
4
5
 => 5 
```

**Comment by zocoi** [comment]
FYI, I end up using `max_by` instead

**Comment by headius** [comment]
Grr. I ported over max and min from MRI but perhaps I messed something up. Will be fixed for 9.1.1.0

**Comment by zocoi** [comment]
thanks @headius
